packages/evolution-legacy/src/components/survey/Section.js

```diff
@@ -166,6 +167,17 @@ export class Section extends React.Component {
         widgetStatus.errorMessage = widgetStatus.errorMessage || serverErrorMessage;
       }
 
+      const join = undefined;
+      if (widgetConfig.joinWith) {
+        const previous = this.props.widgets[i - 1] ? this.props.widgets[i - 1] : undefined;
+        const previousStatus = this.props.widgets[i - 1] ? _get(this.props.interview, `widgets.${this.props.widgets[i - 1]}`, {}) : undefined;
+        const next = this.props.widgets[i + 1] ? this.props.widgets[i + 1] : undefined;
+        const nextStatus = this.props.widgets[i + 1] ? _get(this.props.interview, `widgets.${this.props.widgets[i + 1]}`, {}) : undefined;
+
+        join = widgetConfig.joinWith === previous && previousStatus.isVisible ? 'previous' : undefined;
+        join = widgetConfig.joinWith === next && nextStatus.isVisible? 'next' : undefined;
```

Comment:
You are reassigning the join on this line, so if it's the previous, it will never work. Use
`widgetConfig.joinWith === previous && previousStatus.isVisible ? 'previous' : widgetConfig.joinWith === next && nextStatus.isVisible? 'next' : undefined;`